extensions/interactions/RatioExpressionInput/RatioExpressionInput.py

```diff
@@ -48,8 +48,9 @@ class RatioExpressionInput(base.BaseInteraction):
         }
     }, {
         'name': 'numberOfTerms',
```

Comment:
Are we verifying anywhere that this can’t be 1?

Reply:
No we don't check this because this logic only works if value is greater than 2 otherwise treat this as 0 and don't throw any errors for learner

Reply:
Just to clarify this if we write anything less than 2 for example 0 or 1 in that case if user writes something 1:2:3 this would be valid but if the creator sets the value as 4 in that 1:2:3 would lead to a submit time error in Android. Does that make sense ?
 lso i did have this conversation with sean and we agreed on having the term as 0.

Reply:
You can't treat 1 as 0. That's unclear.

My suggestion would just be, in the logic, to treat 1 as representing a 1-element ratio (no colons). I know that can't happen in practice but that would help make the logic of the check simpler and clearer.

Make sure also that in the editor we forbid creators from specifying 1.

Reply:
In the current scenario if the creator enter's 0(default) or 1 that wouldn't affect anything because current logic works if the number of terms is greater than 2. We could show a description text that entering 0 or 1 won't have any effect. Would that work? @seanlip

Reply:
Also if learners are entering 1: that would give a validation error and won't allow the user to submit the answer in the first place so it doesn't make any sense.

Reply:
@seanlip @BenHenning we could show a warning if the number of terms is 1 but it would be confusing that having 0 is okay but having 1 is not allowed that sounds a bit weird to me.

Reply:
For the 1 case, I meant just "1", not "1:". Wouldn't that work? I'm OK with "1" being considered OK for now, at least in the backend; strictly speaking, it shouldn't mess up any of the checks. You can still have a validation error for it in the editor frontend.

Reply:
nope, that won't work it would throw a frontend validation error (for the creator). If you referring to leaner's flow we could change that but for now, that too does throw a validation error also it would mess up the backend because we have validation of `is_atleast` with a value of 2

Reply:
OK, I think there is some confusion. My suggestion is:

- For creators, have a validation error in the frontend if they enter anything that is not "0" or an integer >= 2.
- For learners, accept ratios that have the correct number of elements as stated by the cust. arg. If the cust. arg value happens to be 1 (which it won't be), just have the validation be that it's a single integer (with no colons). You do not need to check the validity of the cust. arg.
- For the backend, the code that this comment is attached to has a validation of "at least 0", which is correct.

Reply:
For creators, have a validation error in the frontend if they enter anything that is not "0" or an integer >= 2.
you mean integer <=2 correct?

Reply:
... that is not ("0" or an integer >= 2).

Reply:
Thanks for the clarification :) Also for the error message if the value = 1 
`Number of terms cannot be 1`
The above message would be sufficient or should i add some context for the creator?

Reply:
"The number of terms in a ratio should be greater than 1" is sufficient.